File: leetcode/Easy/3768-check-if-digits-are-equal-in-string-after-operations-i/3768-check-if-digits-are-equal-in-string-after-operations-i.cpp
/*
 Problem: 3768. Check If Digits Are Equal in String After Operations I
 Language: cpp
 Runtime: 9 ms (30.79%)
 Memory: 11.6 MB (12.71%)
 Tags: Math, String, Simulation, Combinatorics, Number Theory
*/
class Solution {
public:
    bool hasSameDigits(string s) {
        vector<int> digits;
        digits.reserve(s.size());
        for (char c : s) digits.push_back(c - '0');

        while (digits.size() > 2) {
            vector<int> next;
            next.reserve(digits.size() - 1);
            for (size_t i = 0; i + 1 < digits.size(); ++i) {
                next.push_back((digits[i] + digits[i + 1]) % 10);
            }
            digits.swap(next);
        }

        return digits[0] == digits[1];
    }
};

File: leetcode/Easy/3768-check-if-digits-are-equal-in-string-after-operations-i/3768-check-if-digits-are-equal-in-string-after-operations-i.py
"""
 Problem: 3768. Check If Digits Are Equal in String After Operations I
 Language: python3
 Runtime: 95 ms (5.30%)
 Memory: 17.8 MB (88.84%)
 Tags: Math, String, Simulation, Combinatorics, Number Theory
"""
class Solution:
    def hasSameDigits(self, s: str) -> bool:
        while len(s) > 2:
            new_s = ''
            for i in range(len(s) - 1):
                new_digit = (int(s[i]) + int(s[i + 1])) % 10
                new_s += str(new_digit)
            s = new_s
        return s[0] == s[1]